### ISSUE TITLE
v2 - Cannot switch thermostats on iOS

### ISSUE THREAD

**Issue by Clete2** [ziebelje/beestat-issues#67]
I can't switch thermostats on mobile Safari. [See video](https://share.icloud.com/photos/0KTBUeI5aYQJIvKBrlU9dTKUw)

Works fine on the computer.

I can run Safari devtools if you like.

**Comment by ziebelje** [comment]
Neat...I'll find an iPhone somewhere and get that fixed. Appreciate the video documentation; that's a huge help!

**Comment by ziebelje** [comment]
If you switch thermostats on the computer, then refresh your phone, does the thermostat change? That setting is stored server-side so the refresh should grab the new setting on reload.

**Comment by Clete2** [comment]
Yes it seems to update when I do that.

Plot twist: Switching from the top option (for me, Downstairs) to the bottom option (for me, Upstairs) works. But bottom -> top (up -> down) does not.

**Comment by ziebelje** [comment]
I can't validate and test this until tonight, but my theory is a race condition here:
```
      .addEventListener('click', function() {
        beestat.setting('thermostat_id', thermostat_id);
        window.location.reload();
      });
```
I think the window is reloading before the API call to change the thermostat completes.

**Comment by ziebelje** [comment]
This issue is fixed and will go live with beestat 1.21 sometime this week.

**Comment by Clete2** [comment]
Thanks!!
